File: src/test/java/no/fintlabs/InstanceFlowHeadersMappingServiceTest.java
package no.fintlabs;

import no.fintlabs.flyt.kafka.headers.InstanceFlowHeaders;
import no.fintlabs.mapping.InstanceFlowHeadersMappingService;
import no.fintlabs.repository.entities.InstanceFlowHeadersEmbeddable;
import org.junit.jupiter.api.Test;

import java.util.List;
import java.util.UUID;

import static org.junit.jupiter.api.Assertions.assertEquals;
import static org.junit.jupiter.api.Assertions.assertNotNull;

class InstanceFlowHeadersMappingServiceTest {

    private final InstanceFlowHeadersMappingService mapper = new InstanceFlowHeadersMappingService();

    @Test
    void testToEmbeddable() {
        InstanceFlowHeaders instanceFlowHeaders = InstanceFlowHeaders.builder()
                .sourceApplicationId(1L)
                .sourceApplicationIntegrationId("integrationId")
                .sourceApplicationInstanceId("instanceId")
                .correlationId(UUID.randomUUID())
                .integrationId(1L)
                .instanceId(1L)
                .configurationId(1L)
                .archiveInstanceId("archiveId")
                .build();

        InstanceFlowHeadersEmbeddable embeddable = mapper.toEmbeddable(instanceFlowHeaders);

        assertNotNull(embeddable);
        assertEquals(instanceFlowHeaders.getSourceApplicationId(), embeddable.getSourceApplicationId());
        assertEquals(instanceFlowHeaders.getSourceApplicationIntegrationId(), embeddable.getSourceApplicationIntegrationId());
        assertEquals(instanceFlowHeaders.getSourceApplicationInstanceId(), embeddable.getSourceApplicationInstanceId());
        assertEquals(instanceFlowHeaders.getCorrelationId(), embeddable.getCorrelationId());
        assertEquals(instanceFlowHeaders.getIntegrationId(), embeddable.getIntegrationId());
        assertEquals(instanceFlowHeaders.getInstanceId(), embeddable.getInstanceId());
        assertEquals(instanceFlowHeaders.getConfigurationId(), embeddable.getConfigurationId());
        assertEquals(instanceFlowHeaders.getArchiveInstanceId(), embeddable.getArchiveInstanceId());
    }

    @Test
    void testToInstanceFlowHeaders() {
        InstanceFlowHeadersEmbeddable embeddable = InstanceFlowHeadersEmbeddable.builder()
                .sourceApplicationId(1L)
                .sourceApplicationIntegrationId("integrationId")
                .sourceApplicationInstanceId("instanceId")
                .fileIds(List.of(UUID.randomUUID()))
                .correlationId(UUID.randomUUID())
                .integrationId(1L)
                .instanceId(1L)
                .configurationId(1L)
                .archiveInstanceId("archiveId")
                .build();

        InstanceFlowHeaders instanceFlowHeaders = mapper.toInstanceFlowHeaders(embeddable);

        assertNotNull(instanceFlowHeaders);
        assertEquals(embeddable.getSourceApplicationId(), instanceFlowHeaders.getSourceApplicationId());
        assertEquals(embeddable.getSourceApplicationIntegrationId(), instanceFlowHeaders.getSourceApplicationIntegrationId());
        assertEquals(embeddable.getSourceApplicationInstanceId(), instanceFlowHeaders.getSourceApplicationInstanceId());
        assertEquals(embeddable.getCorrelationId(), instanceFlowHeaders.getCorrelationId());
        assertEquals(embeddable.getFileIds(), instanceFlowHeaders.getFileIds());
        assertEquals(embeddable.getIntegrationId(), instanceFlowHeaders.getIntegrationId());
        assertEquals(embeddable.getInstanceId(), instanceFlowHeaders.getInstanceId());
        assertEquals(embeddable.getConfigurationId(), instanceFlowHeaders.getConfigurationId());
        assertEquals(embeddable.getArchiveInstanceId(), instanceFlowHeaders.getArchiveInstanceId());
    }

}

File: src/test/kotlin/no/novari/flyt/history/InstanceFlowHeadersMappingServiceTest.kt
package no.novari.flyt.history

import no.novari.flyt.history.mapping.InstanceFlowHeadersMappingService
import no.novari.flyt.history.repository.entities.InstanceFlowHeadersEmbeddable
import no.novari.flyt.kafka.instanceflow.headers.InstanceFlowHeaders
import org.junit.jupiter.api.Assertions.assertEquals
import org.junit.jupiter.api.Assertions.assertNotNull
import org.junit.jupiter.api.Test
import java.util.UUID

class InstanceFlowHeadersMappingServiceTest {
    private val mapper = InstanceFlowHeadersMappingService()

    @Test
    fun testToEmbeddable() {
        val instanceFlowHeaders =
            InstanceFlowHeaders
                .builder()
                .sourceApplicationId(1L)
                .sourceApplicationIntegrationId("integrationId")
                .sourceApplicationInstanceId("instanceId")
                .correlationId(UUID.randomUUID())
                .integrationId(1L)
                .instanceId(1L)
                .configurationId(1L)
                .archiveInstanceId("archiveId")
                .build()

        val embeddable = mapper.toEmbeddable(instanceFlowHeaders)

        assertNotNull(embeddable)
        assertEquals(instanceFlowHeaders.sourceApplicationId, embeddable.sourceApplicationId)
        assertEquals(instanceFlowHeaders.sourceApplicationIntegrationId, embeddable.sourceApplicationIntegrationId)
        assertEquals(instanceFlowHeaders.sourceApplicationInstanceId, embeddable.sourceApplicationInstanceId)
        assertEquals(instanceFlowHeaders.correlationId, embeddable.correlationId)
        assertEquals(instanceFlowHeaders.integrationId, embeddable.integrationId)
        assertEquals(instanceFlowHeaders.instanceId, embeddable.instanceId)
        assertEquals(instanceFlowHeaders.configurationId, embeddable.configurationId)
        assertEquals(instanceFlowHeaders.archiveInstanceId, embeddable.archiveInstanceId)
    }

    @Test
    fun testToInstanceFlowHeaders() {
        val embeddable =
            InstanceFlowHeadersEmbeddable
                .builder()
                .sourceApplicationId(1L)
                .sourceApplicationIntegrationId("integrationId")
                .sourceApplicationInstanceId("instanceId")
                .fileIds(listOf(UUID.randomUUID()))
                .correlationId(UUID.randomUUID())
                .integrationId(1L)
                .instanceId(1L)
                .configurationId(1L)
                .archiveInstanceId("archiveId")
                .build()

        val instanceFlowHeaders = mapper.toInstanceFlowHeaders(embeddable)

        assertNotNull(instanceFlowHeaders)
        assertEquals(embeddable.sourceApplicationId, instanceFlowHeaders.sourceApplicationId)
        assertEquals(embeddable.sourceApplicationIntegrationId, instanceFlowHeaders.sourceApplicationIntegrationId)
        assertEquals(embeddable.sourceApplicationInstanceId, instanceFlowHeaders.sourceApplicationInstanceId)
        assertEquals(embeddable.correlationId, instanceFlowHeaders.correlationId)
        assertEquals(embeddable.fileIds, instanceFlowHeaders.fileIds)
        assertEquals(embeddable.integrationId, instanceFlowHeaders.integrationId)
        assertEquals(embeddable.instanceId, instanceFlowHeaders.instanceId)
        assertEquals(embeddable.configurationId, instanceFlowHeaders.configurationId)
        assertEquals(embeddable.archiveInstanceId, instanceFlowHeaders.archiveInstanceId)
    }
}
